### PR TITLE
feat: add component dataTable & state pageActive using pinia

### PR DESCRIPTION
**Pull request by ristianaditya**
## Changes

- add state pageActive
- add dataTable component using NuxtUI

## Preview
image

## Evidence
title: add state pageActive &  add dataTable component using NuxtUI
project: Jabar Super Apps
participants: @rayfajars @johndy2742 @ristianaditya @MNAlfiansyah

### CODE REVIEW

File: dashboard/common/constant/organisasi.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const organisasiHeaders: { label?: string; key: string; sortable?:boolean }[] = [
+    { label: "Organisasi", key: "name", sortable: true },
+    { label: "Email", key: "email", sortable: true },
+    { label: "Address", key: "address" },
+    { label: "Photo", key: "photo" },
+    { key: 'actions' }
+];
```
Comment (Ibwedagama) [inline dashboard/common/constant/organisasi.ts:7]: hello @ristianaditya imho we should use Interface to improve code readability

```ts
interface IOrganizationHeader {
  key: string,
  label?: string,
  sortable?: boolean
}

export const organisasiHeaders: IOrganizationHeader[] = [
  // ...
]

```

File: dashboard/composables/useFetchData.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        query: { page: page.value },
+        onResponse({ response }) {
+          resolve(response._data.data);
+        },
+      });
+    } catch (error) {
+      reject(error);
+    }
+  });
+}
```
Comment (Ibwedagama) [inline dashboard/composables/useFetchData.ts:18]: please add new line on this file @ristianaditya

**Review by Ibwedagama — APPROVED** [review]

**Review by marsellavaleria19 — APPROVED** [review]

**Review by doohanas — APPROVED** [review]

File: dashboard/common/constant/organisasi.ts (hunk 1 above)
Comment (ristianaditya) [reply dashboard/common/constant/organisasi.ts]: hello kang @Ibwedagama, thanks for your advice. I've fixed it. This is my commit : [913d20f](https://github.com/jabardigitalservice/event-management-system/pull/6/commits/913d20faf297e58fe1d830cc7ca4bd33aac35577)

File: dashboard/composables/useFetchData.ts (hunk 2 above)
Comment (ristianaditya) [reply dashboard/composables/useFetchData.ts]: hello kang @Ibwedagama, thanks for your advice. I've fixed it. This is my commit : [https://github.com/jabardigitalservice/event-management-system/pull/6/commits/e0f5b22dd25abe6153db3e05e1d2297ca07143e6](e0f5b22)

**Review by Ibwedagama — APPROVED** [review]

**Review by rayfajars — APPROVED** [review]

**Pull request merged**